### PR TITLE
feat: add names resources

### PR DESCRIPTION
**Pull request by yashin5**
this pr aims to add some resources to names

### CODE REVIEW

File: lib/name_of_the_day/names.ex [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    Name.changeset(name, attrs)
+  end
+
+  def get_name_by(opts) do
+    where = opts |> Enum.reject(fn {_k, v} -> is_nil(v) end)
+
+    Name
+    |> where(^where)
+    |> Repo.all()
+  end
```
Comment (yashin5) [inline lib/name_of_the_day/names.ex:111]: @joaothallis see jão

File: lib/name_of_the_day/names.ex [hunk 2]
```diff
@@ -0,0 +1,112 @@
+defmodule NameOfTheDay.Names do
+  @moduledoc """
+  The Names context.
```
Comment (joaothallis) [inline lib/name_of_the_day/names.ex:3]: @yashin5 what do you thing about create the controllers and HTML resource using `mix phx.gen.html`?

File: test/name_of_the_day/names_test.exs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+defmodule NameOfTheDay.NamesTest do
+  use NameOfTheDay.DataCase
```
Comment (joaothallis) [inline test/name_of_the_day/names_test.exs:2]:
```suggestion
  use NameOfTheDay.DataCase, async: true
```

File: lib/name_of_the_day/names.ex [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+      iex> change_name(name)
+      %Ecto.Changeset{data: %Name{}}
+
+  """
+  def change_name(%Name{} = name, attrs \\ %{}) do
+    Name.changeset(name, attrs)
+  end
+
+  def get_name_by(opts) do
```
Comment (joaothallis) [inline lib/name_of_the_day/names.ex:105]:
```suggestion
  keyword :: list(%Name{})
  def get_name_by(opts) do
```

File: lib/name_of_the_day/names.ex (hunk 2 above)
Comment (yashin5) [reply lib/name_of_the_day/names.ex:3]: thats the idea. i think about first create the core context with phx.gen.context and them after that, use phx.gen.html

**Pull request merged**